xla/service/hlo_computation_deduplicator_test.cc

```diff
@@ -618,5 +618,35 @@ TEST_F(HloComputationDeduplicatorTest, DontDeduplicateReduceAllReduce) {
   auto computation_names = RunDeduplicatePass(text, /*expect_true=*/false);
   EXPECT_EQ(computation_names.size(), 3);
 }
+
+TEST_F(HloComputationDeduplicatorTest, RemoveDuplicateAllReduce) {
+  const std::string_view text = R"(
+  HloModule TestModule
+
+  add.1 {
+    Arg_0 = s32[] parameter(0)
+    Arg_1 = s32[] parameter(1)
+    ROOT add.2 = s32[] add(Arg_0, Arg_1)
+  }
+  add.2 {
+    Arg_0 = s32[] parameter(0)
+    Arg_1 = s32[] parameter(1)
+    ROOT add.2 = s32[] add(Arg_0, Arg_1)
+  }
+
+  ENTRY main {
+    Arg_0.1 = s32[] parameter(0)
+    rd1 = s32[] all-reduce(Arg_0.1), to_apply=add.1
+    Arg_1.1 = s32[] parameter(1)
+    rd2 = s32[] all-reduce(Arg_1.1), to_apply=add.2
+    ROOT multiply.14 = s32[] multiply(rd1, rd2)
+  }
+  )";
+  auto computation_names = RunDeduplicatePass(text, /*expect_true=*/true);
+  EXPECT_EQ(computation_names.size(), 2);
+  for (auto name : computation_names) {
+    EXPECT_NE(name, "add.2");
+  }
+}
```

Comment:
Isn't your change about not deduplicating something that was deduplicated before? If so it's not clear to me what in this test shows the change? Wouldn't this test pass already before your change?

Reply:
Right, my change is indeed about not deduplicating something that was deduplicated before, but as a somewhat of a side effect, it is also deduplicating something that was not deduplicated before (computations of two all-reduces). This unit test is documenting that.

Reply:
> my change is indeed about not deduplicating something that was deduplicated before,

I would expect a test for this though?

Reply:
I have added a test that checks the deduplicator with a wrongly set IsCollectiveCalledComputation flag now. Is that what you had in mind?